--- 0/src/lib/tolua_event.c	2021-11-09 19:34:29.569831100 +0200
+++ 0/src/lib/tolua_event.c	2012-08-24 04:01:53.000000000 +0300
@@ -23,12 +23,12 @@
 static void storeatubox (lua_State* L, int lo)
 {
 	#ifdef LUA_VERSION_NUM
-		lua_getfenv(L, lo);
+		lua_getuservalue(L, lo);
 		if (lua_rawequal(L, -1, TOLUA_NOPEER)) {
 			lua_pop(L, 1);
 			lua_newtable(L);
 			lua_pushvalue(L, -1);
-			lua_setfenv(L, lo);	/* stack: k,v,table  */
+			lua_setuservalue(L, lo);	/* stack: k,v,table  */
 		};
 		lua_insert(L, -3);
 		lua_settable(L, -3); /* on lua 5.1, we trade the "tolua_peers" lookup for a settable call */
@@ -141,7 +141,7 @@
 	{
 		/* Access alternative table */
 		#ifdef LUA_VERSION_NUM /* new macro on version 5.1 */
-		lua_getfenv(L,1);
+	        lua_getuservalue(L,1);
 		if (!lua_rawequal(L, -1, TOLUA_NOPEER)) {
 			lua_pushvalue(L, 2); /* key */
 			lua_gettable(L, -2); /* on lua 5.1, we trade the "tolua_peers" lookup for a gettable call */
--- 0/src/lib/tolua_map.c	2021-11-09 19:38:43.806127700 +0200
+++ 0/src/lib/tolua_map.c	2021-11-09 19:41:37.609538900 +0200
@@ -263,7 +263,7 @@
 		lua_pop(L, 1);
 		lua_pushvalue(L, TOLUA_NOPEER);
 	};
-	lua_setfenv(L, -2);
+	lua_setuservalue(L, -2);
 
 	return 0;
 };
@@ -271,7 +271,7 @@
 static int tolua_bnd_getpeer(lua_State* L) {
 
 	/* stack: userdata */
-	lua_getfenv(L, -1);
+	lua_getuservalue(L, -1);
 	if (lua_rawequal(L, -1, TOLUA_NOPEER)) {
 		lua_pop(L, 1);
 		lua_pushnil(L);
@@ -411,7 +411,7 @@
 		lua_rawget(L,-2);
 	}
 	else
-	 lua_pushvalue(L,LUA_GLOBALSINDEX);
+	 lua_pushglobaltable(L);
 }
 
 /* End module
@@ -445,7 +445,7 @@
 	else
 	{
 		/* global table */
-		lua_pushvalue(L,LUA_GLOBALSINDEX);
+	        lua_pushglobaltable(L);
 	}
 	if (hasvar)
 	{
@@ -473,7 +473,7 @@
 	else
 	{
 		/* global table */
-		lua_pushvalue(L,LUA_GLOBALSINDEX);
+	        lua_pushglobaltable(L);
 	}
 	if (hasvar)
 	{
--- 0/src/lib/tolua_push.c	2021-11-09 19:49:07.234283700 +0200
+++ 0/src/lib/tolua_push.c	2012-08-24 04:01:53.000000000 +0300
@@ -79,7 +79,7 @@
 
    #ifdef LUA_VERSION_NUM
    lua_pushvalue(L, TOLUA_NOPEER);
-   lua_setfenv(L, -2);
+   lua_setuservalue(L, -2);
    #endif
   }
   else
--- 0/src/bin/tolua.c	2012-08-24 04:01:53.000000000 +0300
+++ 0/src/bin/tolua.c	2021-11-09 19:54:18.129722400 +0200
@@ -67,7 +67,7 @@
 static void add_extra (lua_State* L, char* value) {
 	int len;
 	lua_getglobal(L, "_extra_parameters");
-	len = luaL_getn(L, -1);
+	len = lua_rawlen(L, -1);
 	lua_pushstring(L, value);
 	lua_rawseti(L, -2, len+1);
 	lua_pop(L, 1);
